cmd/azure.go

```diff
@@ -0,0 +1,363 @@
+		t, e := time.Parse(time.RFC1123, container.Properties.LastModified)
+		if e != nil {
+			return nil, traceError(e)
+		}
+		buckets = append(buckets, BucketInfo{
+			Name:    container.Name,
+			Created: t,
+		})
+	}
+	return buckets, nil
+}
+
+// DeleteBucket - Use Azure equivalent DeleteContainer.
+func (a AzureObjects) DeleteBucket(bucket string) error {
+	return azureToObjectError(traceError(a.client.DeleteContainer(bucket)), bucket)
+}
+
+// ListObjects - Use Azure equivalent ListBlobs.
+func (a AzureObjects) ListObjects(bucket, prefix, marker, delimiter string, maxKeys int) (result ListObjectsInfo, err error) {
+	resp, err := a.client.ListBlobs(bucket, storage.ListBlobsParameters{
+		Prefix:     prefix,
+		Marker:     marker,
+		Delimiter:  delimiter,
+		MaxResults: uint(maxKeys),
+	})
+	if err != nil {
+		return result, azureToObjectError(traceError(err), bucket, prefix)
+	}
+	result.IsTruncated = resp.NextMarker != ""
+	result.NextMarker = resp.NextMarker
+	for _, object := range resp.Blobs {
+		t, e := time.Parse(time.RFC1123, object.Properties.LastModified)
+		if e != nil {
+			continue
+		}
+		result.Objects = append(result.Objects, ObjectInfo{
+			Bucket:          bucket,
+			Name:            object.Name,
+			ModTime:         t,
+			Size:            object.Properties.ContentLength,
+			MD5Sum:          object.Properties.Etag,
+			ContentType:     object.Properties.ContentType,
+			ContentEncoding: object.Properties.ContentEncoding,
+		})
+	}
+	result.Prefixes = resp.BlobPrefixes
+	return result, nil
+}
+
+// GetObject - Use Azure equivalent GetBlobRange.
+func (a AzureObjects) GetObject(bucket, object string, startOffset int64, length int64, writer io.Writer) (err error) {
+	byteRange := fmt.Sprintf("%d-%d", startOffset, startOffset+length-1)
+	if length == -1 {
+		byteRange = fmt.Sprintf("%d-", startOffset)
+	}
+	rc, err := a.client.GetBlobRange(bucket, object, byteRange, nil)
+	if err != nil {
+		return azureToObjectError(traceError(err), bucket, object)
+	}
+	_, err = io.Copy(writer, rc)
+	rc.Close()
+	return traceError(err)
+}
+
+// GetObjectInfo - Use Azure equivalent GetBlobProperties.
+func (a AzureObjects) GetObjectInfo(bucket, object string) (objInfo ObjectInfo, err error) {
+	prop, err := a.client.GetBlobProperties(bucket, object)
+	if err != nil {
+		return objInfo, azureToObjectError(traceError(err), bucket, object)
+	}
+	t, err := time.Parse(time.RFC1123, prop.LastModified)
+	if err != nil {
+		return objInfo, traceError(err)
+	}
+	objInfo.Bucket = bucket
+	objInfo.ContentEncoding = prop.ContentEncoding
+	objInfo.ContentType = prop.ContentType
+	objInfo.MD5Sum = prop.Etag
+	objInfo.ModTime = t
+	objInfo.Name = object
+	objInfo.Size = prop.ContentLength
+	return
+}
+
+// PutObject - Use Azure equivalent CreateBlockBlobFromReader.
+func (a AzureObjects) PutObject(bucket, object string, size int64, data io.Reader, metadata map[string]string, sha256sum string) (objInfo ObjectInfo, err error) {
```

Comment:
The line wrap issue can be taken care of in a separate fix because:
```
krishna@escape:~/gocode/src/github.com/minio/minio/cmd$ cat * | awk 'length($0) > 100' | wc -l
1996
krishna@escape:~/gocode/src/github.com/minio/minio/cmd$ 

```